src/tasks/migration/DiagnosticProvider.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as _ from "lodash";
+import * as vscode from "vscode";
+import { getTasksJsonPaths } from "./utils";
+import * as readline from "readline";
+import * as fs from "fs";
+import { DeprecatedExportJarTaskProvider, ExportJarTaskProvider } from "../../exportJarSteps/ExportJarTaskProvider";
+
+export class DiagnosticProvider implements vscode.Disposable {
+    public static DIAGNOSTIC_SOURCE = "Project Manager for Java";
+    public static DEPRECATED_TASK_TYPE_DECLARATION = `\"type\": \"${DeprecatedExportJarTaskProvider.type}\"`;
+    public static DEPRECATED_TASK_TYPE_MESSAGE = `Tasks with type \"${DeprecatedExportJarTaskProvider.type}\" are deprecated and will not be supported in the future, please use \"${ExportJarTaskProvider.exportJarType}\" instead.`;
+    private diagnosticCollection: vscode.DiagnosticCollection;
+    private disposables: vscode.Disposable[] = [];
+    private refreshDiagnosticsTrigger: any;
+
+    public dispose() {
+        for (const d of this.disposables) {
+            d.dispose();
+        }
+    }
+
+    public async initialize(): Promise<void> {
```

Comment:
Can we just put the initialization logic into constructor and not await `initializeDiagnostics()`?